services/supervisor/src/lib.rs

```diff
@@ -99,7 +100,10 @@ impl Service for Supervisor {
                         entry.config_propose
                     );
                     // Perform the application of configs.
-                    context.update_config(entry.config_propose.changes);
+                    let communication_channel = context.supervisor_extensions().unwrap();
```

Comment:
Is it possible to cause a panic here? If so, using `expect` with a short description (e.g., `"Blockchain is improperly configured: the `Supervisor` service has incorrect instance ID"`) would be better.